prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -323,6 +324,48 @@ class FhirTransformerTests {
         assertThat(newValue[0].primitiveValue()).isEqualTo("abc123")
     }
 
+    @Test
+    fun `test transform with value set`() {
+        val bundle = Bundle()
+        bundle.id = "bundle1"
+        val resource = Patient()
+        resource.id = "abc123"
+        bundle.addEntry().resource = resource
+        val resource2 = Patient()
+        resource2.id = "def456"
+        bundle.addEntry().resource = resource2
+        val resource3 = Patient()
+        resource3.id = "jkl369"
+        bundle.addEntry().resource = resource3
+
+        val patientElement = FhirTransformSchemaElement(
+            "patientElement",
+            value = listOf("%resource.id"),
+            resource = "%resource",
+            bundleProperty = "%resource.id",
+            valueSet = sortedMapOf(
+                Pair("abc123", "ghi789"),
+                Pair("def456", "")
+            )
+        )
+        val childSchema = FhirTransformSchema(elements = mutableListOf(patientElement))
+
+        val elemA = FhirTransformSchemaElement(
+            "elementA",
+            resource = "Bundle.entry.resource.ofType(Patient)",
+            schemaRef = childSchema,
+            resourceIndex = "patientIndex",
+        )
+
+        val schema = FhirTransformSchema(elements = mutableListOf(elemA))
+
+        FhirTransformer(schema).transform(bundle)
+
+        assertThat(resource.id).isEqualTo("ghi789")
+        assertThat(resource2.id).isNullOrEmpty()
```

Comment:
Is this right? if the pair value is "", should the resource be null?

Reply:
The property should be set to the empty string currently, so this test is less restrictive than we might want it to be. I think we should specifically check for the empty string (which might also boost readability when seen next to the other two assertions here).

```suggestion
        assertThat(resource2.id).isEqualTo("")
```